controller/konnect/konnectextension_controller.go

```diff
@@ -627,6 +648,9 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		&ext,
 		readyCondition,
 	); err != nil || updated || !res.IsZero() {
+		if err != nil && k8serrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
```

Comment:
```suggestion
		if k8serrors.IsNotFound(err) {
			return ctrl.Result{}, nil
		}
```

It's safe to use `nil` with `IsNotFound()`

```
func TestX(t *testing.T) {
	var err error
	require.False(t, k8serrors.IsNotFound(err)) // Passes
}
```